### DISCUSSION TITLE
PnP search v4 Refiner Deep Links

### DISCUSSION THREAD

**Discussion by jameelrrs** [microsoft-search/pnp-modern-search#2217] [Q&A]
Hi, 

Currently we have refiners (PnP Search V4) that allows to be pre-filtered by generating the deep links with the desirable selection on refiner and is able to use from some page as a link.

What I'm trying now is something like below (example deep link):

`f=[{"filterName":"RefinableString01","values":[{"name":"string('abc*')","value":"\"ǂǂ123\"","operator":0,"disabled":false}],"operator":"or"}]#
`
Where, the expectation is to get all the refiners to be pre-select with the search results where **RefinableString01 CONTAINS a value "abc"** .

Example:  If SharePoint documents has a metadata with values, lets say "abc", "abc:123" & "abc:xyz" for each document respectively.  I would like to see all of three in search results with as contains('abc') in the deep link, instead of hardcoding those 3 different values in the deep link as these can be multiple.

Thanks in advance.

**Comment by wobba** [comment]
Not possible the way refiners have been set up to work in the web parts. If you however use your passed in value in the query template, then it will work. And remember that "contains" means that it has the complete term, not that a string contains those characters as KQL only support wildcards at the end.

See https://microsoft-search.github.io/pnp-modern-search/usage/search-results/tokens/#supported-variables and https://docs.microsoft.com/en-us/sharepoint/technical-reference/query-variables.
